### DISCUSSION TITLE
NSPOSIXErrorDomain Code=57 "Socket is not connected"

### DISCUSSION THREAD

**Discussion by wasauce** [deepgram/community#290] [General help]
### Which Deepgram product are you using?

Deepgram API

### Details

Hi - 

We have been streaming results to your API via wss://api.deepgram.com/v1/listen?punctuate=true&alternatives=1&interim_results=true&encoding=linear16&sample_rate=48000&tier=enhanced&model=general&endpointing=500&diarize=true

I see this in our logs:

2023/07/25 07:14:40:677  Decodable.from.data error: keyNotFound(CodingKeys(stringValue: "channel_index", intValue: nil), Swift.DecodingError.Context(codingPath: [], debugDescription: "No value associated with key CodingKeys(stringValue: \"channel_index\", intValue: nil) (\"channel_index\").", underlyingError: nil))
{"type":"Metadata","transaction_key":"deprecated","request_id":"649d781d-82bb-4824-85f5-335f1fc6391a","sha256":"incomplete","created":"2023-07-25T07:14:07.169Z","duration":4.437375,"channels":1,"models":["c7226e9e-ae1c-4057-ae2a-a71a6b0dc588"],"model_info":{"c7226e9e-ae1c-4057-ae2a-a71a6b0dc588":{"name":"general-enhanced","version":"2022-05-18.1","arch":"polaris"}}}

This issue started around midnight last night (so July 25 12:01am Pacific time).

Would love your guidance on what we are doing wrong.

Thank you!

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

wss://api.deepgram.com/v1/listen?punctuate=true&alternatives=1&interim_results=true&encoding=linear16&sample_rate=48000&tier=enhanced&model=general&endpointing=500&diarize=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

649d781d-82bb-4824-85f5-335f1fc6391a

**Comment by jkroll-deepgram** [comment]
Hi @wasauce, I see in Deepgram's logs that we received your streaming query, processed your request, and then did not receive any websocket messages from the driver in 30 seconds, so we closed the stream connection, which would presumably then lead to your error that the socket is not connected. Do you know if there were any changes in your websocket driver, or if that was the end of your streamed data and the connection was just not closed cleanly? Did you observe this once or has it been ongoing since you first observed it?

**Reply by wasauce** [reply]
This is ongoing. We did not change the driver.

**Reply by wasauce** [reply]
@jkroll-deepgram - what does "2023/07/25 07:14:40:677 Decodable.from.data error: keyNotFound(CodingKeys(stringValue: "channel_index", intValue: nil), Swift.DecodingError.Context(codingPath: [], debugDescription: "No value associated with key CodingKeys(stringValue: "channel_index", intValue: nil) ("channel_index").", underlyingError: nil))
{"type":"Metadata","transaction_key":"deprecated","request_id":"649d781d-82bb-4824-85f5-335f1fc6391a","sha256":"incomplete","created":"2023-07-25T07:14:07.169Z","duration":4.437375,"channels":1,"models":["c7226e9e-ae1c-4057-ae2a-a71a6b0dc588"],"model_info":{"c7226e9e-ae1c-4057-ae2a-a71a6b0dc588":{"name":"general-enhanced","version":"2022-05-18.1","arch":"polaris"}}}" mean

We see this each time.

**Reply by jkroll-deepgram** [reply]
It looks like a Swift dictionary error to me - that you are trying to access a `channel_index` key that does not exist. If you look at our [Streaming](https://developers.deepgram.com/reference/streaming#response-schema) API reference, you can see how `channel_index` is one of the response fields.

**Reply by wasauce** [reply]
@jkroll-deepgram ah ok. That makes sense.

The issue has now gone away. We say it at 12:01am, 1am, and then again at about 8am pacific. Today. I just re-tested today at about 11:10am pacific and the issue has stopped. I would like your help understanding the root cause. We have changed nothing with our binary. We tested from multiple locations and repeated the issue at the times I noted. I suspect it was happening at the times between what I marked -- I just cannot personally state that the issue was happening then.
